Python/Product/Analysis/Analyzer/DDG.cs

```diff
@@ -252,53 +252,108 @@ internal class DDG : PythonWalker {
-            } else if (attributes.Length == 0) {
-                throw new ArgumentException("non-empty attributes required", nameof(attributes));
-            }
-            return AssignImportedModuleOrMember(node, userMod, null, attributes, assignName, attributes?.Length == 1);
-        }
-
-        private bool AssignImportedModule(Node node, ModuleReference modRef, IReadOnlyList<string> attributes, string assignName) {
-            return AssignImportedModuleOrMember(node, modRef.Module, modRef.AnalysisModule, attributes, assignName, false);
-        }
-
-        private bool AssignImportedModuleOrMember(Node node, IModule userMod, IAnalysisSet analysisMod, IReadOnlyList<string> attributes, string assignName, bool addLink) {
-            var variable = Scope.CreateVariable(node, _unit, assignName, false);
-            if (userMod == null) {
-                return false;
-            }
-
-            bool added = false;
-            if (attributes == null) {
-                if (analysisMod != null) {
-                    added = Assign(variable, analysisMod, node);
-                }
-            } else {
-                var value = userMod.GetModuleMember(node, _unit, attributes[0], true, Scope, addLink ? assignName : null);
-                foreach (var n in attributes.Skip(1)) {
-                    value = value.GetMember(node, _unit, n);
+        /// <summary>
+        /// Resolves and returns an imported member.
+        /// </summary>
+        /// <param name="module">
+        /// The module to import or import from.
+        /// </param>
+        /// <param name="attribute">
+        /// An optional attribute (already split at the dots) to
+        /// </param>
+        /// <param name="node">
+        /// The import location. If <paramref name="addRef"/> is true,
+        /// this location will be marked as a reference of the imported
+        /// value.
+        /// </param>
+        /// <param name="addRef">
+        /// True to make <paramref name="node"/> a reference to the
+        /// imported member.
+        /// </param>
+        /// <param name="linkToName">
+        /// If not null, the name in the current scope to link to the
+        /// value in the module. This is only used when 
+        /// <paramref name="attribute"/> contains one element.
+        /// </param>
+        /// <returns>The imported member, or null.</returns>
+        private IAnalysisSet GetImportedModuleOrMember(ModuleReference module, IReadOnlyList<string> attribute, Node node, bool addRef, string linkToName) {
+            if (module?.Module == null) {
+                return null;
+            }
+
+            if (attribute == null || attribute.Count == 0) {
+                return module.AnalysisModule;
+            }
+
+            var value = module.Module.GetModuleMember(node, _unit, attribute[0], addRef, Scope, attribute.Count == 1 ? linkToName : null);
+            foreach (var n in attribute.Skip(1)) {
```

Comment:
What attribute is skipped here?

Reply:
The one from the module. Say we are getting `re.RegexFlag.IGNORECASE`. That would result in us importing `re` and attribute would be `['RegexFlag', 'IGNORECASE']`.

We want to get `RegexFlag` directly from the module using `GetModuleMember`, and then use the generalized `GetMember` to get `IGNORECASE` from whatever is returned. So we skip `attribute[0]`.